src/main/java/cz/centrum/haffner/SimpleTrainingDavid/DataTemplates/MetricsInfoDataSet.java

```diff
@@ -0,0 +1,88 @@
+package cz.centrum.haffner.SimpleTrainingDavid.DataTemplates;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class MetricsInfoDataSet {
```

Comment:
Jaký byl důvod pro to vytvořit MetricsInfoDataSet, jako další komponentu, která se vytvoří ve spring kontextu a bude se nulovat před každým "použitím" ?
Řekněme, že příjde požadavek na použití asynchronního RESTu, protože parsování souborů chvíli trvá.
Co by se stalo (a ono by se to s velkou pravděpodobností na produkci stalo) , kdyby přicházely requesty na zpracování souborů rychleji, než by je stíhal parser zparsovat?
Představ si, že zatímco probíhá parsování a inkrementují se countery pro jeden soubor, tak se metoda 
MetricsInfoService#proceedMetricsInfoDataSet() zavolá s dalším souborem - Jaké hodnoty by byly v counterech?

Reply:
předěláno na pevnou vazbu pomocí syntaxe new() + ta data instance se vytváří přímo v parseru (tj. není mu zbytečně předávána)